arangod/Graph/WeightedEnumerator.cpp

```diff
@@ -57,14 +57,15 @@ WeightedEnumerator::WeightedEnumerator(Traverser* traverser, TraverserOptions* o
   _schreier.reserve(32);
 }
 
-void WeightedEnumerator::setStartVertex(arangodb::velocypack::StringRef startVertex) {
-  PathEnumerator::setStartVertex(startVertex);
-
+void WeightedEnumerator::clear() {
   _schreier.clear();
   _schreierIndex = 0;
   _lastReturned = 0;
   _queue.clear();
+}
 
+void WeightedEnumerator::setStartVertex(arangodb::velocypack::StringRef startVertex) {
+  PathEnumerator::setStartVertex(startVertex);
```

Comment:
As I see before - `setStartVertex` implied "clear" but here I don't see `clear` call like in other enumerators. Is this intended?

Reply:
Yes, it's probably better to call `clear()` here as well!